File: app/src/main/java/com/vocaby/app/data/entity/WordDefinitions.java
package com.vocaby.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class WordDefinitions {
    @Embedded
    public Word word;

    @Relation(
            parentColumn = "id",
            entityColumn = "word_id",
            entity = Definition.class
    )

    public List<Definition> definitions;
}

File: app/src/main/java/com/vocaby/app/data/entity/WordDefinitions.kt
package com.vocaby.app.data.entity

import androidx.room.Embedded
import androidx.room.Relation

data class WordDefinitions(
    @Embedded val wordData: Word,
    @Relation(
        parentColumn = "id",
        entityColumn = "word_id",
    )
    val definitions: List<Definition>
)

